lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -462,21 +470,35 @@
         return fallback;
     }
 
-    function GetNumberOption(options, property, minimum, maximum, fallback) {
-        const rawValue = options[property];
-
-        if (typeof rawValue !== 'undefined') {
-            const formattedValue = Internal.ToNumber(rawValue);
-
-            if (isNaN(formattedValue) || formattedValue < minimum || formattedValue > maximum) {
-                platform.raiseOptionValueOutOfRange_3(String(rawValue), String(property), `[${minimum} - ${maximum}]`);
+    /**
+     * Extracts the value of the property named property from the provided options object,
+     * converts it to a Number value, checks whether it is in the allowed range,
+     * and fills in a fallback value if necessary.
+     * 
+     * NOTE: this has known differences compared to the spec GetNumberOption in order to
+     * support more verbose errors. It is more similar to DefaultNumberOption
+     *
+     * ECMA402: #sec-defaultnumberoption
+     *
+     * @param {Object} options user-provided options object
+     * @param {String} property the property we are trying to get off of `options`
+     * @param {Number} minimum minimum allowable value for options[property]
+     * @param {Number} maximum maximum allowable value for options[property]
+     * @param {Number} fallback return value if options[property] is undefined or invalid
+     * @returns {Number}
+     */
+    const GetNumberOption = function (options, property, minimum, maximum, fallback) {
+        let value = options[property];
+        if (value !== undefined) {
```

Comment:
>value !== undefined [](start = 12, length = 19)

(typeof value !== 'undefined') is a JS idiom which is "necessary" for some reason... we should look into this.

Reply:
A quick google suggests that typeof x !== undefined is to make sure that x was declared && undefined or never declared, or a very old school thing to make sure that nobody redefined `undefined` to something else. I don't think either are concerns for us?